src/operator/timeout.ts

```diff
@@ -20,7 +20,8 @@ export function timeout<T>(this: Observable<T>, due: number | Date,
                            scheduler: Scheduler = async): Observable<T> {
   const absoluteTimeout = isDate(due);
   const waitFor = absoluteTimeout ? (+due - scheduler.now()) : Math.abs(<number>due);
-  const error = errorToSend || new TimeoutError();
+  const error = errorToSend instanceof Error ? errorToSend : new TimeoutError(errorToSend);
```

Comment:
That's good point. One concern about allowing custom object is in that case it won't have stack trace to trace down still which this PR try to avoid for custom errors. Maybe instead of allowing error to be any kind limit its type to `Error` would be more simpler approach? (but this causes breaking changes to interface).